anaconda.py

```diff
@@ -455,13 +453,21 @@ def _earlyExceptionHandler(ty, value, traceback):
         time.sleep(10)
         sys.exit(1)
 
+    # Find a kickstart file.
+    kspath = startup_utils.find_kickstart(opts)
+    log.info("Found a kickstart file: %s", kspath)
+
+    # Run %pre scripts.
+    from pyanaconda import kickstart
+    kickstart.preScriptPass(kspath)
+
     # Collect all addon paths
+    from pyanaconda.addons import collect_addon_paths
     addon_paths = collect_addon_paths(constants.ADDON_PATHS)
 
-    # If we were given a kickstart file on the command line, parse (but do not
-    # execute) that now.  Otherwise, load in defaults from kickstart files
-    # shipped with the installation media.
-    ksdata = startup_utils.parse_kickstart(opts, addon_paths, pass_to_boss=True)
+    # Parse the kickstart file.
+    ksdata = kickstart.AnacondaKSHandler(addon_paths["ks"])
+    kickstart.parseKickstart(ksdata, kspath, strict_mode=opts.ksstrict, pass_to_boss=True)
```

Comment:
Could you please make a function from this part. It will be much readable to see that these parts are connected.

Reply:
Fixed.